tracks/javascript/exercises/pascals-triangle/mentoring.md

```diff
@@ -1,47 +1,44 @@
 ### Reasonable solutions
 
 ```javascript
-function buildRow(result, element, index, array) {
-  return result.concat([element + (array[index + 1] || 0)])
+const buildRow = (result, element, index, array) => {
+  return [...result, element + (array[index + 1] || 0)];
 }
 
-function makeTriangle(levels) {
-  const result = [[1]]
-  for (let x = 1; x < levels; x += 1) {
-    result.push(result[x - 1].reduce(buildRow, [1]))
-  }
-  return result
-}
+export const rows = levels => {
+  if (levels === 0)
+    return [];
 
-export class Triangle {
-  constructor(levels) {
-    this.rows = makeTriangle(levels)
-  }
-
-  get lastRow() {
-    return this.rows[this.rows.length - 1]
+  const result = [[1]]
+  for (let x = 1; x < levels; x++) {
+    result.push(result[x - 1].reduce(buildRow, [1]));
   }
+  return result;
 }
 ```
 
 ```javascript
-export class Triangle {
-    constructor(size) {
-        this.rows = [[1]]
-
-        for (let y = 1; y < size; y++) {
-            this.lastRow = []
-
-            for (let x = 0; x <= y; x++) {
-                const up = this.rows[y - 1]
-                const left = up[x - 1] || 0
-                const right = up[x] || 0
-
-                this.lastRow.push(left + right)
-            }
-            this.rows.push(this.lastRow)
-        }
+export const rows = size => {
+  if (size === 0)
+    return [];
+
+  const result = [[1]];
+
+  for (let y = 1; y < size; y++) {
+    let lastRow = [];
```

Comment:
```suggestion
    const lastRow = [];
```

Reply:
Not sure I agree with that... Have you some rules / guidelines about that?

Personally, I use `const` when I don't want to change this variable (by reference or value)... This works more as an indicator that we should not touch this variable, this way I know that its value is the one at initialization...

What are your thoughts about that?

Reply:
The standard `eslint` configuration (`eslint/recommended`) will recommend to use a `const` when the _binding_ never changes. That in line with what `const` means in JavaScript context. 

The whole "but it confuses me because const doesn't mean constant" or "I want to indicate that a variable should never change" to me is just not a good idea and confuses people even more. The counter argument is that when you have a variable declaration on line 1, you won't know on line 50 anymore if you used `const` or `let`. If It's _vital_ that it doesn't change, `const` is going to help you (but `TypeScript` might, and `Object.freeze` will help).

Reply:
If you change this, we can get this merged!